cli/apiserver/provider.go

```diff
@@ -0,0 +1,167 @@
+package apiserver
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/up9inc/mizu/cli/logger"
+	"github.com/up9inc/mizu/cli/uiUtils"
+	"github.com/up9inc/mizu/shared"
+	"io/ioutil"
+	core "k8s.io/api/core/v1"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type apiServerProvider struct {
+	url     string
+	isReady bool
+}
+
+var Provider = apiServerProvider{}
+
+func (provider *apiServerProvider) Init(url string, retries int) error {
+	healthUrl := fmt.Sprintf("%s/", url)
+	retriesLeft := retries
+	for retriesLeft > 0 {
+		if response, err := http.Get(healthUrl); err != nil {
+			logger.Log.Debugf("[ERROR] failed connecting to api server %v", err)
+		} else if response.StatusCode != 200 {
+			logger.Log.Debugf("can't connect to api server yet, response status code %v", response.StatusCode)
+		} else {
+			logger.Log.Debugf("connection test to api server passed successfully")
+			break
+		}
+		retriesLeft -= 1
+		time.Sleep(time.Second)
+	}
+
+	if retriesLeft == 0 {
+		return fmt.Errorf("couldn't reach the api server after %v retries", retries)
```

Comment:
do we want to return nil here? 
or maybe just change the isReady = false (I guess it is the default but maybe it is better to be specific and set the value to false?)

Reply:
why return nil? it returns error because it failed to connect. the function sets isReady=True when it actually ready (all other cases doing nothing